src/ClimaLSM.jl

```diff
@@ -103,22 +104,89 @@ Returns the component names of the `land` model, by calling
 """
 land_components(land::AbstractLandModel) = propertynames(land)
 
+function prognostic_vars(land::AbstractLandModel)
+    components = land_components(land)
+    prognostic_list = map(components) do model
+        prognostic_vars(getproperty(land, model))
+    end
+    return NamedTuple{components}(prognostic_list)
+end
+
+function prognostic_types(land::AbstractLandModel)
+    components = land_components(land)
+    prognostic_list = map(components) do model
+        prognostic_types(getproperty(land, model))
+    end
+    return NamedTuple{components}(prognostic_list)
+end
+
+function auxiliary_vars(land::AbstractLandModel)
+    components = land_components(land)
+    auxiliary_list = map(components) do model
+        auxiliary_vars(getproperty(land, model))
+    end
+    return NamedTuple{(components..., :interactions)}((
+        auxiliary_list...,
+        interactions_vars(land),
+    ))
+end
+
+function auxiliary_types(land::AbstractLandModel)
+    components = land_components(land)
+    auxiliary_list = map(components) do model
+        auxiliary_types(getproperty(land, model))
+    end
+    return NamedTuple{(components..., :interactions)}((
+        auxiliary_list...,
+        interactions_types(land),
+    ))
+end
```

Comment:
@kmdeck This extends `prognostic_vars` and `prognostic_types` (and for auxiliary) to `AbstractLandModel`s. It returns a named tuple that nests the prognostic/aux vars for each submodel. In the case of auxiliary vars, the interaction variables are also included (non-nested, just like the state/aux containers).